New `Select` decomposition: unary iterator (#7623)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

<h3>New features since last release</h3>

* A new decomposition based on unary iteration was added to :class:`qml.Select`.

>   It requires :math:`c-1` auxiliary wires for a ``Select`` template with :math:`c` control wires,

Reviewer: ```suggestion
  This new decomposition requires :math:`c-1` auxiliary wires for a ``Select`` operation with :math:`c` control wires,
```

>   but reduces the :class:`T` count significantly. The implementation makes use of a simplification

Reviewer: ```suggestion
  but reduces the :class:`T` count significantly. The implementation uses a simplification
```

>   for partial Select templates, which only is valid if the state on the control wires has no

Reviewer: ```suggestion
  for partial ``Select`` templates, which is only valid if the state on the control wires has no
```

>   overlap with computational basis states that are not used by ``Select``. See the documentation

Reviewer: ```suggestion
  overlap with computational basis states not used by ``Select``. See the documentation
```

>   for more details.

Reviewer: Not sure I understand what the second sentence says.

Reviewer 2: It's an attempt at making the following statement less technical: The implementation makes use of a simplification that is only valid if the state $|\psi\rangle$ on the control qubits satisfies $\langle j | \psi\rangle=0$ for all $j\in[K, 2^c-1)$, where $K$ is the number of operators applied by ``Select``, and $c$ is the number of control wires.

Reviewer 3: I will add a suggestion. I don't think we necessarily need to include those details, and I see some other explanations that we've added to the docs which would work better here.

>   for more details.

Reviewer 2: The text is very nice, thanks @AntonNI8. Just to make sure: Users will not find the condition, under which the new decomposition is correct, in the changelog, right?
